Features

issamelimrany opened the pull request:
In this PR I've tried to remove as much of duplication as possible ( creation of helpers.js and a hooks folder ), and created the unit test for the helper functions. 

![tests](https://github.com/user-attachments/assets/eb5988c1-afea-4021-b800-3b88b2f00e96)

https://github.com/user-attachments/assets/049a84f5-84c1-44de-b1de-d96b8b8e3e6e

LaghzaliTaha commented on components/hooks/useDecimalInput.ts:
```diff
@@ -0,0 +1,21 @@
+  setValue(new Decimal(value));
+};
+
+const useDecimalInput = (initialValue: Decimal = new Decimal(0)) => {
```
i see that you love duplication , why you didnt use useFormInput 

const useDecimalInput = (initialValue: Decimal = new Decimal(0)) => {
  const [value, handleChange] = useInput({ value: initialValue }, value => new Decimal(value));
  return [value.value, handleChange] as const;
};

issamelimrany commented:
I've extracted the displaying part of the forms into 2 components, and reverted to old implementation of useDecimalInput ( the new one is commented, it does compile correctly, but doesn't update the state in a correct manner )

LaghzaliTaha commented on components/displayedResult.tsx:
```diff
@@ -0,0 +1,27 @@
+    <>
+      {results.map((result, index) => (
+        <p key={index} className="text-sm text-gray-700">
+          {result.label}:{" "}
```
don't use this syntax {result.label}:{" "} instead use template literals
